src/theory/partition_generator.h

```diff
@@ -0,0 +1,124 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * PartitionGenerator for creating partitions.
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__THEORY__SPLITTER_H
+#define CVC5__THEORY__SPLITTER_H
+
+#include <vector>
+
+#include "proof/trust_node.h"
+#include "smt/env_obj.h"
+#include "theory/theory.h"
+#include "theory/valuation.h"
+
+namespace cvc5 {
+
+class TheoryEngine;
+
+namespace prop {
+class PropEngine;
+}
+
+namespace theory {
+
+class PartitionGenerator : protected EnvObj
+{
+ public:
+  PartitionGenerator(Env& env,
+                     TheoryEngine* theoryEngine,
+                     prop::PropEngine* propEngine);
+
+  /**
+   * Make partitions for parallel solving. e communicates the effort at which makePartitions was called.
+   * Returns a lemma blocking off the emitted cube from the search.
+   */
+  TrustNode makePartitions(Theory::Effort e);
```

Comment:
I think we should simply call this method `check`, as we are not directly asking it to make a partition, but rather check what it should do.

Reply:
Do you think that it would make this confusing in theory_engine.cpp? 

https://github.com/amaleewilson/cvc5/blob/dbe6b72db744b3e314def8b6fcdbd2711bb83cb3/src/theory/theory_engine.cpp#L406

```
    if (d_partitionGen != nullptr)
    {
      TrustNode tl = d_partitionGen->makePartitions(effort);
      if (!tl.isNull()){
        lemma(tl, LemmaProperty::NONE, THEORY_LAST);
      }
    }
```

Reply:
No, I think it would make it clearer. At a high level, the method we are calling is similar to a theory solver, whose main entry point is also called `check`.